src/runtime_src/core/tools/common/tests/TestBandwidthKernel.cpp

```diff
@@ -25,8 +36,241 @@ TestBandwidthKernel::run(std::shared_ptr<xrt_core::device> dev)
     ptree.put("status", test_token_failed);
     return ptree;
   }
-  std::string testcase = (name.find("vck5000") != std::string::npos) ? "versal_23_bandwidth.py" : "23_bandwidth.py";
-  runTestCase(dev, testcase, ptree);
-  
+  runTest(dev, ptree);
   return ptree;
 }
+
+static bool 
+is_emulation() {
+  bool ret = false;
+  char* xcl_mode = std::getenv("XCL_EMULATION_MODE");
+  if (xcl_mode != nullptr) {
+    ret = true;
+  }
+  return ret;
+}
+
+void
+TestBandwidthKernel::runTest(std::shared_ptr<xrt_core::device> dev, boost::property_tree::ptree& ptree)
+{
+  const auto bdf_tuple = xrt_core::device_query<xrt_core::query::pcie_bdf>(dev);
+  const std::string bdf = xrt_core::query::pcie_bdf::to_string(bdf_tuple);
+  const std::string test_path = findPlatformPath(dev, ptree);
```

Comment:
I guess this is used with platform.json? Please define it closer to the actual implementation